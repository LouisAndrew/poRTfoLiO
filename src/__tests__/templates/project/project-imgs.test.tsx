import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

// import { render, cleanup } from '@testing-library/react'
import { cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'

import ProjectImgs from 'templates/project/project-imgs'

// export type Screenshot = {
//     screenshotLabel: string
//     screenshot: {
//         childImageSharp: {
//             fluid: FluidObject
//         }
//     }
//     screenshotMobile: {
//         childImageSharp: {
//             fluid: FluidObject
//         }
//     }
// }

describe('ProjectImgs', () => {
    const Element = (
        <ProjectImgs screenshots={[]} gifs={{ gif: '', gifMobile: '' }} />
    )

    afterEach(cleanup)

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(Element, div)
    })

    /* it('renders correctly', () => {
		const { getByTestId } = render()
	}) */

    it('matches snapshot', () => {
        const run = true

        expect(run).toBeTruthy()

        if (run) {
            const tree = renderer.create(Element).toJSON()
            expect(tree).toMatchSnapshot()
        }
    })
})
