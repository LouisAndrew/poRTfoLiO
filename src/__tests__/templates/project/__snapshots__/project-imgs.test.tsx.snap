// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectImgs matches snapshot 1`] = `
<div
  className="css-1353cfc"
>
  <div
    className="carousel"
  >
    <div
      aria-live="polite"
      className="horizontalSlider___281Ls carousel__slider carousel__slider--horizontal carousel-slide"
      onKeyDown={[Function]}
      role="listbox"
      style={Object {}}
      tabIndex={0}
    >
      <div
        className="carousel__slider-tray-wrapper carousel__slider-tray-wrap--horizontal"
        style={Object {}}
      >
        <ul
          className="sliderTray___-vHFQ sliderAnimation___300FY carousel__slider-tray carousel__slider-tray--horizontal"
          onClickCapture={[Function]}
          onMouseDown={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "flexDirection": "row",
              "transform": "translateX(0%) translateX(0px)",
              "width": "100%",
            }
          }
        >
          <li
            aria-selected={true}
            className="slide___3-Nqo slideHorizontal___1NzNV carousel__slide carousel__slide--visible"
            onBlur={[Function]}
            onFocus={[Function]}
            role="option"
            style={
              Object {
                "paddingBottom": "66.69921875%",
                "width": "100%",
              }
            }
            tabIndex={0}
          >
            <div
              className="slideInner___2mfX9 carousel__inner-slide slide"
              style={Object {}}
            >
              <div
                className="css-y9gkef"
              >
                <video
                  autoPlay={true}
                  className="desktop-gif"
                  loop={true}
                  muted={true}
                  playsInline={true}
                >
                  <source
                    src=""
                    type="video/mp4"
                  />
                </video>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
    <div
      className="carousel__dot-group carousel-dot"
    >
      <button
        aria-label="slide dot"
        className="dot___3c3SI carousel__dot carousel__dot--0 carousel__dot--selected"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        <span />
      </button>
    </div>
  </div>
</div>
`;
